src/WebJobs.Script/FileProvisioning/PowerShell/PowerShellFileProvisioner.cs

```diff
@@ -30,22 +44,125 @@ public Task ProvisionFiles(string scriptRootPath)
 
         private void AddRequirementsFile(string scriptRootPath)
         {
-            string requirementsFilePath = Path.Combine(scriptRootPath, "requirements.psd1");
+            _logger.LogInformation($"Creating {RequirementsPsd1FileName}.");
+            string requirementsFilePath = Path.Combine(scriptRootPath, RequirementsPsd1FileName);
+
             if (!File.Exists(requirementsFilePath))
             {
-                string content = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
-                File.WriteAllText(requirementsFilePath, content);
+                string requirementsContent = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
+                string guidance = null;
+
+                try
+                {
+                    string majorVersion = GetLatestAzModuleMajorVersion();
+
+                    requirementsContent = requirementsContent.Replace("# 'Az'", "'Az'");
```

Comment:
Please use regex instead of string replace, this would eliminate the possibilities of not taking into account white spaces e.g.
     string matchPattern = "(\\s?)*#(\\s?)*'Az'";
            var text = "    #            'Az' = 'MAJOR_VERSION.*'";

            if (System.Text.RegularExpressions.Regex.IsMatch(text, matchPattern, System.Text.RegularExpressions.RegexOptions.IgnoreCase))
            {
                var modified_text = Regex.Replace(text, matchPattern, "'Az'");
            }

Reply:
I did a research on this, and regex replace is way faster.  Thank you for the suggestion. Code updated.